### PR TITLE
Refactor to enable support for optional JDBC parameters for all JDBC destinations

### LINKED ISSUE #10593: Generalized solution for jdbc parameters
We want to support arbitrary jdbc parameters for all jdbc sources and destinations.
more context in https://github.com/airbytehq/airbyte/issues/9501

### PR DESCRIPTION
**Pull request by girarda**
## What
* Refactor `AbstractJdbcDestination` to streamline support for optional JDBC parameters
* Next step will be to actually add support for all JDBC destinations

## How
1. Add an abstract method
`protected abstract Map<String, String> getDefaultConnectionProperties(final JsonNode config);` 
which concrete destination can implement to define default parameters that need to be set.

2. When building a connection, parse the jdbc url parameters and add the parameters to the default ones, making sure the default parameters are not being overridden.
3. Create the Database using a property map instead of a formatting string. This step is not strictly required, but makes the flow easier to follow.

## Recommended reading order
1. `airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java`: This is where the core of the logic is. The new code is pretty much just moved from `MySQLDestination` as a way to generalize the solution
2. `airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestination.java`: A lot of code was moved to `AbstractJdbcDestination` and `allowLoadLocalInfile`.
3. `airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java` implements the jdbc parameters parsing to a Map<String, String>
4. `airbyte-db/lib/src/main/java/io/airbyte/db/Databases.java`: Removed the option to pass in connectionProperties as a String in favor of always passing a Map<String, String>
5. `airbyte-integrations/connectors/destination-clickhouse/src/main/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestination.java`: Pass default parameters as connection properties instead of embedding them in the URL
6. `airbyte-integrations/connectors/destination-mariadb-columnstore/src/main/java/io/airbyte/integrations/destination/mariadb_columnstore/MariadbColumnstoreDestination.java`: Pass default parameters as connection properties instead of embedding them in the URL
7. `airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java`: Pass default parameters as connection properties instead of embedding them in the URL
8. `airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftInsertDestination.java`:  Pass default parameters as connection properties instead of embedding them in the URL
9. `airbyte-integrations/connectors/destination-mssql/src/main/java/io/airbyte/integrations/destination/mssql/MSSQLDestination.java`: Pass default parameters as connection properties instead of embedding them in the URL
10. `airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java`: Pass default parameters as connection properties instead of embedding them in the URL
11. `airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInsertDestination.java`: Implement `getDefaultConnectionProperties`
12. `airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java`: Implement `getDefaultConnectionProperties`
13. `airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java`: parse connection_properties string to a map of properties before calling `createStreamingJdbcDatabase`

In particular, I want to point out of the following:

- `SnowflakeInsertDestination` and `SnowflakeInternalStagingDestination` don't follow the same pattern as the other connectors
- MSSQLDestination appears to have a bug (see https://github.com/airbytehq/airbyte/pull/10421/files#diff-2143cdde6ae6b3ccd558438e64155a7b5c7199101de227997a472a4e3f6014b8R49)

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [x] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftCopyDestinationAcceptanceTest.java [hunk 1]
```diff
@@ -59,9 +60,19 @@ protected JsonNode getFailCheckConfig() {
...
+        .map(j -> {
+          final String text = j.get(JavaBaseConstants.COLUMN_NAME_DATA).asText();
+          final JsonNode deserialized = Jsons.deserialize(text);
+          if (deserialized.isTextual()) {
+            final JsonNode d = Jsons.deserialize(deserialized.asText());
+            return d;
+          } else {
+            return deserialized;
+          }
+        })
```
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftCopyDestinationAcceptanceTest.java:75]: Did we manage to identify a change in redshift destination behavior that would make the string being serialized twice??

**Review by ChristopheDuong — APPROVED** [review]

**Comment by girarda** [comment]
/publish connector=connectors/destination-mariadb-columnstore

**Comment by girarda** [comment]
/publish connector=connectors/destination-mariadb-columnstore

**Comment by kzzzr** [comment]
Hey @girarda,
Let's say for instance, for Clickhouse Destination I'd like to supply `socket_timeout=300000` JDBC option (which is 30000 = 30sec by default).

How do I supply it via creating destination? There's no such node in `spec.json` for now.

To finalize tables after sync one needs to insert into destination tables from `_tmp` tables. If table is big enough a query might be timed out -> sync fails.

**Comment by girarda** [comment]
@kzzzr we still need to resolve https://github.com/airbytehq/airbyte/issues/10717 before we can start passing additional parameters to Clickhouse destination.

It's going to work similarly to how it works for the MySQL destination - there will be a new field in the spec `jdbc_url_params` where you'll be able to set `socket_timeout=300000`

File: airbyte-db/lib/src/main/java/io/airbyte/db/Databases.java [hunk 2]
```diff
@@ -131,9 +131,9 @@ public static Database createDatabase(final String username,
                                         final String jdbcConnectionString,
                                         final String driverClassName,
                                         final SQLDialect dialect,
-                                        final String connectionProperties) {
+                                        final Map<String, String> connectionProperties) {
```
Comment (ChristopheDuong) [inline airbyte-db/lib/src/main/java/io/airbyte/db/Databases.java:134]: Are there some needed updates on this code https://github.com/airbytehq/airbyte/blob/c7b8db702e8c0505f624101905ef778a12ae10f0/airbyte-integrations/connectors/source-oracle-strict-encrypt/src/test-integration/java/io/airbyte/integrations/source/oracle_strict_encrypt/OracleStrictEncryptSourceAcceptanceTest.java#L63 due to these changes?

**Pull request merged**